ironfish/src/blockchain/nullifierSet/nullifierSet.ts

```diff
@@ -0,0 +1,132 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+import { Assert } from '../../assert'
+import { Block, Transaction } from '../../primitives'
+import { Nullifier } from '../../primitives/nullifier'
+import {
+  BUFFER_ENCODING,
+  DatabaseSchema,
+  IDatabase,
+  IDatabaseStore,
+  IDatabaseTransaction,
+  StringEncoding,
+  U32_ENCODING,
+} from '../../storage'
+import { NullifierInfo, NullifierLocationEncoding } from './encoding'
+
+interface CounterEntry<T extends string> extends DatabaseSchema {
```

Comment:
This doesn't matter too much but using interfaces for serialization is a little dangerous because it's permissive. It's better to use types if you ever loop over the keys.

Reply:
I don't think we can `extend` types that's why we have interface here. We do this for all other DBSchema fields in [schema.ts](https://github.com/iron-fish/ironfish/blob/97ab6751b0b698f8e05674b282e6b04ce3ea12d1/ironfish/src/blockchain/schema.ts) as well.